app/src/main/java/com/github/swent/echo/data/repository/RepositoryImpl.kt

```diff
@@ -236,6 +237,27 @@ class RepositoryImpl(
         remoteDataSource.deleteUserProfile(userProfile)
         localDataSource.deleteUserProfile(userProfile.userId)
     }
+
+    override suspend fun getUserProfilePicture(userId: String): File? {
+        if (isOffline()) {
+            return null
+        }
+        return remoteDataSource.getUserProfilePicture(userId)
+    }
```

Comment:
I think it would make sense to cache images locally on disk (to avoid network delay). Additionally, this would allow to serve images from cache even if offline.

The following documentation/tutorial could help implementing a LRU disk cache for images: https://developer.android.com/topic/performance/graphics/cache-bitmap

An even simpler way, as you already store the images to the cache directory, could be to simply check there if the picture already exists and when it has been saved. If it exists and that has not been too long ago, return the image from local storage directly, otherwise fetch from supabase. 

Also when offline, you could still see if there is a cached copy of the image available and return that.

You can take a look at how I did the caching/synchronisation for the database records above :)

Reply:
These ideas are good. I already tried to save the images on a specific cache sub-folder, but I had issues with the permissions and even the app cache path returned by `context.cacheDir` wasn't writable. The only way I could achieve to  save a file into the cache is by using the default cache directory of the method `File .createTempFile`. The problem with this solution is that I don't know where the default cache directory is, so I can't look into this directory before fetching the picture from Supabase.
I could address these issues in a future sprint, but given the time constraint I prefer to keep it like this even if it's not an optimal solution. Thanks for pointing to some helpful documentation to solve this problem.